app/modals/reset-wallet/reset-wallet-modal.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { FC, useState, useRef } from 'react';
+import { Modal, ButtonGroup, Button, Box } from '@blockstack/ui';
+import { TxModalHeader, TxModalFooter } from '../transaction/transaction-modal-layout';
+import { remote } from 'electron';
+import { clearDiskStorage } from '../../utils/disk-store';
+import { useHotkeys } from 'react-hotkeys-hook';
+
+interface ResetWalletModalProps {
+  isOpen: boolean;
+  onClose(): void;
+}
+
+export const ResetWalletModal: FC<ResetWalletModalProps> = ({ isOpen, onClose }) => {
+  const [wipingWallet, setWipingWallet] = useState(false);
+  const timer = useRef<number>(0);
+  const cancelBtnRef = useRef<HTMLDivElement>();
+  const PANIC_CANCEL_TIME = 3500;
+
+  const closeModal = () => {
+    setWipingWallet(false);
+    clearTimeout(timer.current);
+    onClose();
+  };
+
+  useHotkeys('esc', closeModal);
+
+  const resetWallet = () => {
+    setWipingWallet(true);
+    // Allow user to grace period to panic cancel operations
+    // Focusing cancel btn ensures any key press of: enter, space, esc
+    // will cancel the pending operation
+    cancelBtnRef.current?.focus();
+    timer.current = setTimeout(() => {
+      clearDiskStorage();
+      remote.getCurrentWindow().reload();
```

Comment:
Is this necessary? It seems lead to the lagging blank screen. I would imagine Redux could handle this, potentially better?

Reply:
I did it this way thinking that it's just a little bit safer in the respect of reducing the chance there's some persisted state.

The white space during reload isn't good though. [Tracking in this issue.](https://github.com/blockstack/stacks-wallet/issues/224)